src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectAuthenticationHandler.cs

```diff
@@ -217,6 +235,35 @@ public OpenIdConnectAuthenticationHandler(HttpClient backchannel)
 
             message.State = Options.StateDataFormat.Protect(properties);
 
+            if (Options.UsePostAuthorizationRequests)
+            {
+                using (var buffer = new MemoryStream())
+                using (var writer = new StreamWriter(buffer))
+                {
+                    var inputs = new StringBuilder();
+                    foreach (var parameter in message.Parameters)
+                    {
+                        inputs.AppendFormat(InputTagFormat, parameter.Key, parameter.Value);
+                    }
+
+                    writer.Write(string.Format(CultureInfo.InvariantCulture, HtmlFormat, message.IssuerAddress, inputs));
+                    writer.Flush();
+
+                    Response.ContentLength = buffer.Length;
```

Comment:
@Tratcher I used a local buffer to avoid chunking. I'm fine removing it if you don't like it.